email/parse.c

```diff
@@ -1076,35 +1076,40 @@ int mutt_rfc822_parse_line(struct Envelope *env, struct Email *e, const char *na
 /**
  * mutt_rfc822_read_line - Read a header line from a file
  * @param fp      File to read from
- * @param line    Buffer to store the result
- * @param linelen Length of buffer
- * @retval ptr Line read from file
+ * @param out     Buffer to store the result
+ * @retval num Number of bytes read from fp
  *
  * Reads an arbitrarily long header field, and looks ahead for continuation
- * lines.  "line" must point to a dynamically allocated string; it is
- * increased if more space is required to fit the whole line.
+ * lines.
  */
-char *mutt_rfc822_read_line(FILE *fp, char *line, size_t *linelen)
+size_t mutt_rfc822_read_line(FILE *fp, struct Buffer *out)
 {
-  if (!fp || !line || !linelen)
-    return NULL;
+  if (!fp || !out)
+    return 0;
+
+  mutt_buffer_seek(out, 0);
 
+  size_t linelen = 1024;
+  char *line = mutt_mem_malloc(linelen);
   char *buf = line;
   int ch;
   size_t offset = 0;
+  size_t read = 0;
 
   while (true)
   {
-    if (!fgets(buf, *linelen - offset, fp) || /* end of file or */
+    if (!fgets(buf, linelen - offset, fp) || /* end of file or */
         (IS_SPACE(*line) && !offset))         /* end of headers */
```

Comment:
In this case, you don't update read. It needs to be done before checking IS_SPACE() (I do this in my patch).